plugins/module_utils/ds8000.py

```diff
@@ -0,0 +1,197 @@
+        return [object_type['name'] for object_type in parsed_response]
+
+    def get_volume_ids_from_name(self, volume_name):
+        volume_ids = []
+        volumes = self.get_all_volumes_in_ds8000_storage()
+        for volume in volumes:
+            if volume['name'] == volume_name:
+                volume_ids.append(volume['id'])
+        return volume_ids
+
+    def get_ds8000_objects_by_type(self, sub_url, ds8000_object_type):
+        ds8000_objects = []
+        response = custom_get_request(self.module, self.headers, sub_url)
+        parsed_response = json.loads(response.text)['data'][ds8000_object_type]
+        for obj in parsed_response:
+            ds8000_objects.append({
+                "name": obj['name'],
+                "id": obj['id']
+            })
+        return ds8000_objects
+
+
+    def connect_to_api(self):
+        module = self.module
+        hostname = module.params['hostname']
+        username = module.params['username']
+        password = module.params['password']
+    
+        if not hostname:
+            module.fail_json(msg="Hostname parameter is missing."
+                                 " Please specify this parameter in task")
+    
+        if not username:
+            module.fail_json(msg="Username parameter is missing."
+                                 " Please specify this parameter in task")
+    
+        if not password:
+            module.fail_json(msg="Password parameter is missing."
+                                 " Please specify this parameter in task")
+    
+        rest_client = Client(
+            service_address=hostname,
+            user=username,
+            password=password)
+        return rest_client
+    
+    
+    def get_rest_api_token(self):
+        module = self.module
+        hostname = module.params['hostname']
+        username = module.params['username']
+        password = module.params['password']
+        port = module.params['port']
+        schema = module.params['http_schema']
+        validate_certs = module.params['validate_certs']
+        auth_url = '/tokens'
+        headers = {'Content-Type': 'application/json'}
+        data = '{request:{params:{username:' + username + ',password:' + password + '}}}'
+    
+        response = requests.post(
+            '{schema}://{hostname}:{port}{base_url}{auth}'.format(
+                schema=schema,
+                hostname=hostname,
+                port=port,
+                base_url=DEFAULT_BASE_URL,
+                auth=auth_url),
+            headers=headers,
+            data=data,
+            verify=validate_certs)
+        if response.ok:
+            token = get_the_rest_api_token_from_the_reponse(response.text)
+            return token
+        module.fail_json(msg="failed to get the rest api token:"
+                             " {response}".format(response=response.text))
+        return None
+
+
+def get_the_rest_api_token_from_the_reponse(response):
```

Comment:
Put inside the class (and other functions below as well as long as we call it from "inside" the class/its subclasses.
Rename:
get_token_from_response